app/src/app/classes/stix/stix-object.ts

```diff
@@ -565,6 +564,103 @@ export abstract class StixObject extends Serializable {
+          count = organizationNamespace['range_start'];
+          count = (Number(count) > 0 ? count : 1).toString().padStart(4, '0'); // make sure ID has 4 digits i.e. 0 -> 0001 (note: padStart() is unsupported in IE)
+        }
+        let accessor = this.attackType == "group" ? restAPIConnector.getAllGroups() :
+                       this.attackType == "mitigation" ? restAPIConnector.getAllMitigations() :
+                       this.attackType == "software" ? restAPIConnector.getAllSoftware() :
+                       this.attackType == "tactic" ? restAPIConnector.getAllTactics() :
+                       this.attackType == "technique" ? restAPIConnector.getAllTechniques() :
+                       this.attackType == "data-source" ? restAPIConnector.getAllDataSources() :
+                       this.attackType == "matrix" ? restAPIConnector.getAllMatrices() : null;
+        // Find all other objects that have this prefix and range, and set ID to the most recent and unique ID possible
+        if (accessor) {
+          prefix += this.attackIDValidator.format.includes('#') ? this.attackIDValidator.format.split('#')[0] : '';
+          return accessor.pipe(map((objects) => {
+            let substr = prefix + count.substr(0,3); // i.e. look for 'PRE-T123x' matches
+            let filtered = objects['data'].filter((obj) => obj.attackID.startsWith(substr));
+
+            let reg = new RegExp("\\d{4}");
+            filtered = filtered.sort((a, b,) => {
+              a = a.attackID.match(reg);
+              b = b.attackID.match(reg);
+              return a[0] - b[0] // check which 4-digit ID is greater
+            });
+
+            let latest = filtered.length > 0 ? filtered.pop().attackID.match(reg)[0] : '0';
+            latest = Number(latest)
+            count = (Number(count) > latest ? count : latest + 1).toString().padStart(4, '0');
+
+            // Generate next available ID (assumes user had already clicked generate once and that's why the attackID already exists)
+            if (copyID && copyID.replace(/[A-Z]+-/i,'').endsWith(count)) {
+              count = (Number(count) + 1).toString().padStart(4, '0');
+            }
+
+            if (this.hasOwnProperty('is_subtechnique') && this['is_subtechnique']) {
+              if (this.hasOwnProperty('parentTechnique') && this['parentTechnique']) {
+                let children$ = restAPIConnector.getTechnique(this['parentTechnique'].stixID, null, "all");
+                if (children$) {
+                  let sub = children$.subscribe({
+                    next: t => {
+                      if (t[0] && t[0].attackID) {
+                        count = t[0].attackID.replace(/[A-Z]+-/, "").replace(/[A-Z]/, ''); // 'PRE-T1234' -> '1234'
+                        count += '.001';
+                        let reg = new RegExp("[.]\\d{2}");
+                        let children = t[0].subTechniques;
```

Comment:
It looks like you're generating the sub-technique ordinal from the `subtechnqiues` field on the parent technique, but this field isn't populated without the `includeSubs` parameter on the REST API call. The call just needs to be updated to include that param:
```
let children$ = restAPIConnector.getTechnique(this['parentTechnique'].stixID, null, "latest", true);
```